lib/pos_block_producer.go

```diff
@@ -0,0 +1,145 @@
+package lib
```

Comment:
I was told to hold off on reviewing this file so going to wait.

Reply:
The time has come